merkledag/utils/utils.go

```diff
@@ -212,8 +212,9 @@ func copyDag(nd *dag.ProtoNode, from, to dag.DAGService) error {
 		}
 
 		childpb, ok := child.(*dag.ProtoNode)
-		if !ok {
-			return dag.ErrNotProtobuf
+		if !ok { // leaf node
```

Comment:
I think this is a bit risky to do this way. Lets switch on the type, and then explicitly handle the ProtoNode and RawNodes. Then for everything else we can error out.

Reply:
Why not allow linking to other types? I actually need this to link to git commits, it works well for my usecase (ipld git helper). (note that it's already possible to do this with dag commands)

Reply:
The issue isnt that we don't want to allow linking, but that this operation is supposed to recursively copy a graph from one dagservice to another.

Reply:
@magik6k we can change the first argument of this function to accept a `node.Node`, then skip this check entirely

Reply:
done